src/index/CompressedBlockPrefiltering.cpp

```diff
@@ -0,0 +1,170 @@
+//  Copyright 2024, University of Freiburg,
+//                  Chair of Algorithms and Data Structures
+//  Author: Hannes Baumann <baumannh@informatik.uni-freiburg.de>
+
+#include "index/CompressedBlockPrefiltering.h"
+
+namespace prefilterExpressions {
+
+// SECTION RELATIONAL OPERATIONS
+//______________________________________________________________________________
+template <CompOp Comparison>
+bool RelationalExpressions<Comparison>::compareImpl(
+    const ValueId& firstId, const ValueId& secondId) const {
+  using enum CompOp;
+  // For compactness we combine LE and GT here. In evaluate (see below), for
+  // LE we select all blocks at a smaller (lower_bound) index position, and
+  // for GT all the respective blocks at larger or equal indices.
+  if constexpr (Comparison == LE || Comparison == GT) {
+    return compareIds(firstId, secondId, LE) ==
+           valueIdComparators::ComparisonResult::True;
+  } else {
```

Comment:
I think this doesn't quite work yet, because you have to use the `getRangesForId` function, not simply use `compareIds`. (this function also has a comment, so you should be able to use it without much trouble).

The reason is, that the internal sorting (which is used on the blocks) groups all ints and all doubles together, and also some of them are in inverted order (positives before negatives) etc. 
Probably You didn't notice this, because your implementation seems to work, if your blocks for testing are also sorted incorrectly (not according to the simply operator< instead of `compareIds`. We probably should add an assertion to the `BlockMetadata` that the first triple is less than the last triple to avoid such confusion for future implementers.

The consequence is, that you will have multiple ranges (a small number of them), not a single range of upper and lower bound.

Reply:
Should properly use getRangesForId() now.